documentation/add/11_sqw_compression.md

```diff
@@ -0,0 +1,201 @@
+		- direct and inverse geometry instruments (i.e. Mushroom)
+		- Need to be able to generate these dummy data on demand using developmnet versions of the compressed format to test timing improvements during development.
+	- ‘real-world’ cuts from ‘real-world’ sqw files
+		- Benchmarking and profiling on IDAaaS and local machines
+		- Lots of small cuts 
+		- Large cuts
+
+Alex and Jacob have done some of benchmarking and profiling: Alex on total timings on IDAaaS and local machines, Jacob some testing of times for I/O .v. on-the-fly computation of qx,qy,qz,en using calculate_qw_pixels2.m. What is needed is definitive, documented, and reproducible test examples. 
+
+Profiling is important because past experience with accessing unique object storage, and of accessing detector and run information in Tobyfit, has shown that without careful optimisation these operations can become very time-consuming.
+
+
+### Why store the signal and variance as float64?
+The contents on disk and in memory will then be identical. A problem we have had is comparing data when debugging because pixels close to the edge of a bin can move from one bin to another, as well as the obvious 7 sig. fig. rounding that takes place on double to single conversion.
+
+The problem of pixel-to-bin mapping should disappear with on-the-fly computation of qx,qy,qz,en, as this will be done in memory from the run-detector-energy bin indices. That leaves just the question of the signal and error. Rounding to 7 sig fig may not be important. This should be considered further; particularly if the fraction of non-zero signal pixels is large for a significant proportion of experiments, then saving signal and error as float64 might result in a significant penalty (see Table 2). This will certainly be the case when simulating data with a model, when typically *all* pixels will have non-zero simulated signal. In this case the compressed format file will be 50% larger with float64 signal compared to float32 (and 2/3 the size of the current sqw file compared to 44% for float32).
+
+
+### The meaning of ‘non-zero signal’
+By non-zero signal we actually mean zero signal and zero error: nothing was counted. This is distinct from the case of a background subtraction that has been performed in the data reduction such that there is zero signal and non-zero error. 
+
+
+### Values of nrun, ndet_max, ne_max
+In practice, we won’t always know the values of nrun, ndet_max, ne_max in advance of construction of an sqw file, for example, if the sqw file is built up during an experiment by accumulating the data of newly arrived nxspe files as they appear. In practice, however, we can use default values that will be larger than any feasible experiment in the lifetime of ISIS or other facilities:
+
+|          | Estimate of maximum value | Value | Comment
+|-----------------------------|-----------------|-----------------------------
```

Comment:
You need additional `|` in this line to match the line above and below, e.g.

```
|----------|---------------------------|-------|---------
```

Reply:
See comment above; likewise.